src/Aquarius.Client/Samples/Client/SamplesClient.cs

```diff
@@ -0,0 +1,233 @@
+        {
+            InvokeWebServiceMethod(() => _client.Put(requestDto));
+        }
+
+        public TResponse Delete<TResponse>(IReturn<TResponse> requestDto)
+        {
+            return InvokeWebServiceMethod(() => _client.Delete(requestDto));
+        }
+
+        public void Delete(IReturnVoid requestDto)
+        {
+            InvokeWebServiceMethod(() => _client.Delete(requestDto));
+        }
+
+        public TResponse SparsePut<TResponse>(IReturn<TResponse> requestDto)
+        {
+            return InvokeWebServiceMethod(() => _client.Put(requestDto), WithSparsePutScope);
+        }
+
+        public void SparsePut(IReturnVoid requestDto)
+        {
+            InvokeWebServiceMethod(() => _client.Put(requestDto), WithSparsePutScope);
+        }
+
+        public TResponse PostFileWithRequest<TResponse>(string path, IReturn<TResponse> requestDto)
+        {
+            var fileToUpload = new FileInfo(path);
+
+            using (var stream = fileToUpload.OpenRead())
+            {
+                return PostFileWithRequest(stream, fileToUpload.Name, requestDto);
+            }
+        }
+
+        public TResponse PostFileWithRequest<TResponse>(Stream contentToUpload, string uploadedFileName, IReturn<TResponse> requestDto)
+        {
+            var fileUploader = new FileUploader(_client);
+
+            return InvokeWebServiceMethod(() => fileUploader.PostFileWithRequest(GetPostUrl(requestDto), contentToUpload, uploadedFileName, requestDto));
+        }
+
+        private string GetPostUrl<TResponse>(IReturn<TResponse> requestDto)
+        {
+            return _client.ResolveTypedUrl(HttpMethods.Post, requestDto);
+        }
+
+        public LazyResult<TDomainObject> LazyGet<TDomainObject, TRequest, TResponse>(TRequest requestDto)
+            where TRequest : IPaginatedRequest, IReturn<TResponse>
+            where TResponse : IPaginatedResponse<TDomainObject>
+        {
+            var responseDto = Get(requestDto);
+
+            return new LazyResult<TDomainObject>
+            {
+                TotalCount = responseDto.TotalCount,
+                DomainObjects = GetPaginatedResults<TDomainObject, TRequest, TResponse>(requestDto, responseDto)
+            };
+        }
+
+        private IEnumerable<TDomainObject> GetPaginatedResults<TDomainObject, TRequest, TResponse>(TRequest requestDto, TResponse responseDto)
+            where TRequest : IPaginatedRequest, IReturn<TResponse>
+            where TResponse : IPaginatedResponse<TDomainObject>
+        {
+            var totalCount = responseDto.TotalCount;
+
+            for (var count = 0; count < totalCount;)
+            {
+                foreach (var domainObject in responseDto.DomainObjects)
+                {
+                    yield return domainObject;
+                }
+
+                count += responseDto.DomainObjects.Count;
+
+                if (count >= totalCount || !responseDto.DomainObjects.Any())
```

Comment:
Depending on the source of the list, the total count could change between requests. This isn't a very safe mechanism to find the end of the list anyway.

I'd suggest testing if either the last request returned no objects, or no cursor to determine if you're at the end of the list.